src/vs/workbench/parts/terminal/electron-browser/terminalProcess.js

```diff
@@ -48,6 +48,7 @@ ptyProcess.on('data', function (data) {
 });
 
 ptyProcess.on('exit', function (exitCode) {
+	ptyProcess.kill();
```

Comment:
Thanks @chirag64!

Does everything seem to work fine? It's worth testing on Windows if you haven't. I was thinking we might need a timeout here before we kill `terminalProcess` or it could mess with the pty processes' shutdown.